build_sdk/sdk-release/sscapi.h

```diff
@@ -108,6 +108,8 @@ typedef int ssc_bool_t;
 #define SSC_ARRAY 3
 #define SSC_MATRIX 4
 #define SSC_TABLE 5
+#define SSC_DATARR 6
+#define SSC_DATMAT 7
```

Comment:
Make sure that these are fully supported in SAM - casewin.cpp line 1349, throughout code generator for all languages and in ide.cpp line 399, 416 and invoke.cpp line 1505 and 1624 and through simulation.cpp - 601,697,773, 798,964

Reply:
Thanks for pointing these out! Currently these new data types aren't used as SAM variables or SSC variables. No compute module uses these, and there are no UI widgets to create these. 

The only place they're used is in the reopt equation to store [an array of tables](https://github.com/NREL/ssc/blob/8cc46aed2f898f91b5e8502372053df48e720fae/ssc/cmod_pvsamv1_eqns.cpp#L245) for some of the utility rate fields such as demand rate structure. 

[These functions](https://github.com/NREL/SAM/blob/e0594b30a5c115f8436a4b8ae06ad39096f0ad22/src/variables.h#L47) in `variables.cpp` take care of translating from ssc data (including ssc_datarr, datmat) into lk's var_data and sam's VarValues.

If we wanted to use them in SAM, then we'd have to change `code_generator`, `simulation`, `invoke`. 

Do you think we should add support for these new ssc data types for the release or postpone until we'll actually use them?